setup.py

```diff
@@ -305,6 +311,10 @@ def invariant_checks(cls) -> str:
         """
         raise NotImplementedError()
 
+    @classmethod
+    def implement_optimizations(cls, functions: Dict[str, str]) -> Dict[str, str]:
+        return functions
```

Comment:
Not 100% if I did it in the correct pattern but I intended to make `SpecBuilder` a pure `ABC` class with only `abstractmethod`s. Do you mind (1) moving this method to `Phase0SpecBuilder` and then (2) adding an `abstractmethod` `implement_optimizations` here which raises `raise NotImplementedError()`?

Reply:
sure!